geomstats/geometry/spd_matrices_space.py

```diff
@@ -388,3 +390,10 @@ def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
             base_point = gs.tile(
                 base_point,
                 (gs.maximum(n_tangent_vecs_a, n_tangent_vecs_b), 1, 1))
+
+        aux_a = SPDMatricesSpace(dim).differential_power(2, tangent_vec_a,
```

Comment:
two lines:
- one defining the space: spd_space = SPDMatricesSpace(dim)
- one doing the computation
will make the code faster and easier to read

Reply:
inverse_differential_power as a separate method

Reply:
if you have a better variable name than aux_a that'd be great, otherwise it's fine